### PR TITLE
Fixits for @noreturn -> Never (SE-0102)

### PR DESCRIPTION
**Pull request by slavapestov**
This pull request introduces migration from @noreturn to returning Never, and removes old code for the @noreturn attribute now that it has become an error.

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by nkcsgexi** [comment]
Cool! Thanks for making migration easy for users, Slava!

**Comment by slavapestov** [comment]
@swift-ci Please test and merge

**Comment by slavapestov** [comment]
@swift-ci Please test

File: docs/SIL.rst [hunk 1]
```diff
@@ -1216,8 +1216,8 @@ flow, such as a non-``Void`` function failing to ``return`` a value, or a
 ``switch`` statement failing to cover all possible values of its subject.
 The guaranteed dead code elimination pass can eliminate truly unreachable
 basic blocks, or ``unreachable`` instructions may be dominated by applications
-of ``@noreturn`` functions. An ``unreachable`` instruction that survives
-guaranteed DCE and is not immediately preceded by a ``@noreturn``
+of functions returning uninhabited types. An ``unreachable`` instruction that
+survives guaranteed DCE and is not immediately preceded by an no-return
```
Comment (jtbandes) [inline docs/SIL.rst]: s/an/a/ ?

**Pull request merged**